src/main/java/lotto/controller/AbstractRetry.java

```diff
@@ -0,0 +1,36 @@
+package lotto.controller;
+
+import static lotto.constant.ErrorMessage.EXCEEDING_MAX_RETRY;
+
+import java.util.Objects;
+import java.util.function.Supplier;
+import lotto.constant.MaxRetry;
+
+public abstract class AbstractRetry {
+    public <T> T run(final Supplier<T> supplier) {
+        Objects.requireNonNull(supplier);
+        int retryCount = 0;
+        while (MaxRetry.COUNT.isBelow(retryCount)) {
+            try {
+                return supplier.get();
+            } catch (Exception e) {
+                System.out.println(e.getMessage());
```

Comment:
console에 관련된 로직을 view쪽으로 가져가는건 어떤가요?

Reply:
retry 로직을 나중에 쓸 수 있다고 생각해서 추상화하다보니까 sysout을 사용했었네요. 추상 메서드에서 뷰의 메서드를 참조할 수 있으면 좋은데 자식 클래스의 메서드를 사용할 수 있는 방법이 있을까요? 아니면 추상화를 하지 말고 run메서드를 뷰에 넣는게 좋을까요? @uhanuu 현우님은 어떻게 생각하시는지 궁금합니다.

Reply:
```suggestion
                printErrorMessage(e.getMessage());
            }
            //...
                
protected abstract void handleException(Exception e);
}
```

gpt의 도움을 받아보니 추상메서드를 만들고 자식 메서드에서 자식 메서드를 사용하도록 구현하면 될 거 같아요!

Reply:
추상 클래스의 확장의 개념보다는 File입출력과 Console을 대비한 구현이 맞다고 생각했어요!
- 그래서 저는 default메서드 이용해서 처리했어요! 그리고 View쪽에서 예외는 `IllegalArgumentException`가 대부분이라고 생각했고 Exception은 범위가 너무 큰거 같아서 `IllegalArgumentException` 부모인 `RuntimeException`을 파라미터로 받게 줄여줬어요! 커스텀 예외를 쓴다 하더라도 보통 `RuntimeException`을 상속받아서 사용하니까요